src/main/scala/mesosphere/marathon/DeprecatedFeatures.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon
+
+import com.typesafe.scalalogging.StrictLogging
+
+object DeprecatedFeatures extends StrictLogging {
+  case class DeprecatedFeature(
+      key: String,
+      description: String,
+      warnVersion: SemVer,
+      removeVersion: SemVer
+  )
+
+  val syncProxy = DeprecatedFeature(
+    "sync_proxy",
+    description = "Old, blocking IO implementation for leader proxy used by Marathon standby instances.",
+    warnVersion = SemVer(1, 7, 0),
+    removeVersion = SemVer(1, 8, 0))
+
+  def all = Seq(syncProxy)
+
+  /**
+    * Provided a list of enabled deprecated features, output appropriate log messages based on current version and
+    * deprecation / removal versions.
+    *
+    * @param deprecatedfeature The deprecated features specified to enable
+    * @param currentVersion The current build version
+    * @return True if a a deprecatedFeature was specified that has been specified as removed.
```

Comment:
Heh, wording was confusing but there is an interpretation (which I intended) that is not wrong.

Reply:
It is terribly confusing... it may make sense to you as you know what you mean... however write is intended for the reader not the writer and it do NOT make sense to me.

Reply:
Oh, yeah :) Totally agreed. I had already rewritten it.